apiAuditLogs/apiAuditLogs.go

```diff
@@ -0,0 +1,290 @@
+		"getWebappPlugins":                     true,
+		"listAutocompleteCommands":             true,
+		"listCommands":                         true,
+		"logout":                               true,
+		"openDialog":                           true,
+		"patchChannelModerations":              true,
+		"pinPost":                              true,
+		"postLog":                              true,
+		"pushNotificationAck":                  true,
+		"saveReaction":                         true,
+		"searchAllChannels":                    true,
+		"searchArchivedChannelsForTeam":        true,
+		"searchChannelsForTeam":                true,
+		"searchEmojis":                         true,
+		"searchGroupChannels":                  true,
+		"searchPosts":                          true,
+		"searchTeams":                          true,
+		"searchUserAccessTokens":               true,
+		"searchUsers":                          true,
+		"setPostUnread":                        true,
+		"submitDialog":                         true,
+		"teamExists":                           true,
+		"teamMembersMinusGroupMembers":         true,
+		"testElasticsearch":                    true,
+		"testEmail":                            true,
+		"testLdap":                             true,
+		"testS3":                               true,
+		"testSiteURL":                          true,
+		"unpinPost":                            true,
+		"updateUserStatus":                     true,
+		"uploadFileStream":                     true,
+		"viewChannel":                          true,
+	}
+
+	if pass.Pkg.Path() != "github.com/mattermost/mattermost-server/v5/api4" {
+		return nil, nil
+	}
+
+	inspect := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	nodeFilter := []ast.Node{
+		(*ast.FuncDecl)(nil),
+	}
+	inspect.Preorder(nodeFilter, func(n ast.Node) {
+		funDecl := n.(*ast.FuncDecl)
+
+		if strings.HasSuffix(pass.Fset.File(n.Pos()).Name(), "_test.go") {
+			return
+		}
+		if strings.HasSuffix(pass.Fset.File(n.Pos()).Name(), "apitestlib.go") {
+			return
+		}
+		if obj, ok := pass.TypesInfo.Defs[funDecl.Name].(*types.Func); ok {
+			var fact facts.IsApiHandler
+			if !pass.ImportObjectFact(obj, &fact) {
+				return
+			}
+		}
+		if whiteList[funDecl.Name.Name] {
+			return
+		}
+		initializationFound := false
+		logCallFound := false
+		successCallFound := false
+		ast.Inspect(funDecl, func(node ast.Node) bool {
+			switch n := node.(type) {
+			case *ast.CallExpr:
+				fun, ok := n.Fun.(*ast.SelectorExpr)
+				if !ok {
+					return true
+				}
+
+				ident, ok := fun.X.(*ast.Ident)
+				if !ok {
+					return true
+				}
+
+				// must have a auditRec.Success()
+				if ident.Name == "auditRec" && fun.Sel.Name == "Success" {
+					successCallFound = true
+				}
+				if ident.Name == "c" && (fun.Sel.Name == "LogAuditRec" || fun.Sel.Name == "LogAuditRecWithLevel") {
+					logCallFound = true
+				}
+
+				if ident.Name == "c" && fun.Sel.Name == "MakeAuditRecord" {
```

Comment:
Same as comment above.  `c` is the identifier name for Context by convention, but it seems to be a limitation of the AST tree that we can't do something like "check for identifier of type `web.Context` that calls `MakeAuditRecord`".

Reply:
We can get the name of the first parameter of the handler to check that.

I was trying to have it working and good enough for check all the cases, but we can improve it a bit.

Reply:
Done! :)